src/command_modules/azure-cli-network/azure/cli/command_modules/network/custom.py

```diff
@@ -3600,6 +3608,13 @@ def update_subnet(cmd, instance, resource_group_name, address_prefix=None, netwo
         else:
             instance.address_prefix = address_prefix
 
+    if nat_gateway:
+        instance.nat_gateway = SubResource(id=resource_id(
```

Comment:
Same. This should just be `SubResource(id=nat_gateway)` and the validator should ensure that `nat_gateway` contains an ID.